### ISSUE TITLE
Allow more control over the header (THEAD), default implementation of column searching

### ISSUE THREAD

**Issue by cah-briangantzler** [miguelcobain/ember-yeti-table#14]
Currently the header component contains the thead and tr elements. In order to move the filter from the footer as you have in the example to the header you have to do the code below. This requires a deeper knowledge of the actual implementation.

Would you be opposed to a pull request that provides the ability to add additional rows to the header component.

Since most filters are just an input (sometimes a select), would you be opposed to a pull request that 
provided some kind of a default search implementation (the filter inputs are very boilerplate)

```
<div class="docs-flex docs-justify-end">
  <input
    class="input" type="search" placeholder="Search..."
    value={{filterText}} oninput={{action (mut filterText) value="target.value"}}>
</div>

<YetiTable @data={{data}} @filter={{filterText}} as |table|>

  <thead>

    <table.header @tagName="" as |header|>
        <header.column @prop="firstName" @filter={{firstNameFilter}}>
          First name
        </header.column>
        <header.column @prop="lastName" @filter={{lastNameFilter}}>
          Last name
        </header.column>
        <header.column @prop="points" @filter={{pointsFilter}}>
          Points
        </header.column>
    </table.header>

    <tr>
      <td>
        <input
          class="input" type="search" placeholder="Search first name"
          value={{firstNameFilter}} oninput={{action (mut firstNameFilter) value="target.value"}}>
      </td>
      <td>
        <input
          class="input" type="search" placeholder="Search last name"
          value={{lastNameFilter}} oninput={{action (mut lastNameFilter) value="target.value"}}>
      </td>
      <td>
        <input
          class="input" type="search" placeholder="Search points"
          value={{pointsFilter}} oninput={{action (mut pointsFilter) value="target.value"}}>
      </td>
    </tr>
  </thead>

  <table.body/>

</YetiTable>
```

**Comment by cah-briangantzler** [comment]
I ended up providing an alternate header while still preserving the old way for backward compatibility.

Below is the way filters can be provided.  The columnFilter=true had to the there because the old version did not allow for an optional header and still remain backward compatible. 
```
      <YetiTable @data={{data}} as |table|>

        <table.header @columnFilter={{true}} as |header|>
          <header.columns as |columns|>
            <columns.column @prop="firstName" data-column="test-column">
              First name
            </columns.column>
            <columns.column @prop="lastName">
              Last name
            </columns.column>
            <columns.column @visible={{visibleTest}} @prop="points">
              Points
            </columns.column>
          </header.columns>
          
          <header.filter/>
          
        </table.header>

        <table.body/>

      </YetiTable>
```

Ability to control the look of the filter

```
      <YetiTable @data={{data}} as |table|>

        <table.header @columnFilter={{true}} as |header|>
          <header.columns as |columns|>
            <columns.column @prop="firstName" data-column="test-column">
              First name
            </columns.column>
            <columns.column @prop="lastName">
              Last name
            </columns.column>
            <columns.column @visible={{visibleTest}} @prop="points">
              Points
            </columns.column>
          </header.columns>
          
          <header.filter @placeholder="Search me" as |filter|>
            <filter.column @placeholder="Search on first name"/>
            <filter.column as |filterColumn|>
                <input
                  class="input" type="search" placeholder={{filterColumn.placeholder}}
                  value={{filterColumn.column.filter}}
                  oninput={{action (mut filterColumn.column.filter) value="target.value"}}>
            </filter.column>
            <filter.column/>
          </header.filter>
          
        </table.header>

        <table.body/>

      </YetiTable>
```

**Comment by miguelcobain** [comment]
Closed by #15
